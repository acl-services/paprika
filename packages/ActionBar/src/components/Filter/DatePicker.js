import React from "react";
import PropTypes from "prop-types";
import PaprikaDatePicker from "@paprika/date-picker";
import moment from "moment";

const propTypes = {
  initialDate: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  parsingFormat: PropTypes.string.isRequired,
};

export default function DatePicker(props) {
  const { initialDate, onChange, parsingFormat } = props;

  function getMomentDate(dateStr) {
    return dateStr === "" ? null : moment(dateStr, parsingFormat);
  }

  const memorizedGetMomentDate = React.useCallback(getMomentDate, [parsingFormat]);
  const [date, setDate] = React.useState(memorizedGetMomentDate(initialDate));

  React.useEffect(() => {
    setDate(memorizedGetMomentDate(initialDate));
  }, [initialDate, parsingFormat, memorizedGetMomentDate]);

  function handleChange(newDate) {
    setDate(newDate);
    onChange(newDate);
  }

  return <PaprikaDatePicker onChange={handleChange} date={date} />;
}

DatePicker.propTypes = propTypes;
