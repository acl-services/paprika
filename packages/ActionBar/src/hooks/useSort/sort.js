import moment from "moment";
import getAlphaSortFunction from "./getAlphaSortFunction";

export function compareString(a, b, { locale }) {
  const alphaSortAsc = getAlphaSortFunction(locale);
  return alphaSortAsc(a, b);
}

export function compareNumber(a, b) {
  return a - b;
}

export function compareDate(a, b, { momentParsingFormat }) {
  return moment(a, momentParsingFormat).unix() - moment(b, momentParsingFormat).unix();
}

const compareFunctions = {
  DATE: compareDate,
  NUMBER: compareNumber,
  TEXT: compareString,
  BOOLEAN: compareString,
};

export default function sort({ data, columnId, direction, columnType, momentParsingFormat, locale }) {
  const isAscend = direction === "ASCEND";
  const indicator = isAscend ? 1 : -1;

  function getSortedData(fn) {
    return [...data].sort((rowA, rowB) => {
      return fn(rowA[columnId], rowB[columnId], { momentParsingFormat, locale }) * indicator;
    });
  }

  if (compareFunctions[columnType]) {
    return getSortedData(compareFunctions[columnType]);
  }

  return data;
}
