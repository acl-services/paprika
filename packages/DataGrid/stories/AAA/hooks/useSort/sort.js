import moment from "moment";

function compareString(a, b) {
  return a.localeCompare(b);
}

function compareNumber(a, b) {
  return a - b;
}

function compareDate(a, b, momentParsingFormat) {
  return moment(a, momentParsingFormat).unix() - moment(b, momentParsingFormat).unix();
}

const compareFunctions = {
  DATE: compareDate,
  NUMBER: compareNumber,
  TEXT: compareString,
  BOOLEAN: compareString,
};

export default function sort({ data, columnId, direction, columnType, momentParsingFormat }) {
  const isAscend = direction === "ASCEND";
  const indicator = isAscend ? 1 : -1;

  function getSortedData(fn) {
    return [...data].sort((rowA, rowB) => {
      return fn(rowA[columnId], rowB[columnId], momentParsingFormat) * indicator;
    });
  }

  if (compareFunctions[columnType]) {
    return getSortedData(compareFunctions[columnType]);
  }

  return data;
}
