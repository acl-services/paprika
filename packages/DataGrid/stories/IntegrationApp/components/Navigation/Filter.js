import React from "react";
import nanoid from "nanoid";
import { Filter } from "@paprika/navigation";

const initialValueByType = {
  BOOLEAN: true,
  NUMBER: "",
  TEXT: "",
  DATE: "",
  SINGLE_SELECT: null,
};

const getLevelFilter = () => {
  return {
    columnId: "level",
    rule: "IS",
    value: null,
    filterId: nanoid(),
    renderValueField: () => {
      return (
        <select>
          <option value="low">low</option>
          <option value="mid">mid</option>
          <option value="high">high</option>
        </select>
      );
    },
  };
};

export default function MyFilter(props) {
  const { filters, setFilters, columns, setOperator, operator } = props;

  function getDefaultFilter() {
    return {
      columnId: columns[0].id,
      rule: "IS",
      value: "",
      filterId: nanoid(),
    };
  }

  function handleFilterChange({ filter, rule, value, columnId }) {
    let newFilter;

    if (columnId) {
      const columnType = columns.find(column => column.id === columnId).type;
      if (columnType === "SINGLE_SELECT") {
        newFilter = getLevelFilter();
      } else {
        newFilter = {
          ...filter,
          columnId,
          rule: Filter.rulesByType[columnType][0],
          value: initialValueByType[columnType],
          renderValueField: null,
        };
      }
    } else if (rule) {
      newFilter = {
        ...filter,
        rule,
        value: "",
      };
    } else {
      newFilter = {
        ...filter,
        value,
      };
    }
    setFilters(prevFilters =>
      prevFilters.map(filterItem => (filterItem.filterId === filter.filterId ? newFilter : filterItem))
    );
  }

  const memorizedHandleChange = React.useCallback(handleFilterChange, [columns, setFilters]);

  function handleAddFilter() {
    setFilters(prevFilters => [...prevFilters, getDefaultFilter()]);
  }

  const memorizedAddFilter = React.useCallback(handleAddFilter, [columns, setFilters]);

  function handleDeleteFilter(deletedFilter) {
    setFilters(prevFilters => [...prevFilters].filter(filter => filter.filterId !== deletedFilter.filterId));
  }

  const memorizedDeleteFilter = React.useCallback(handleDeleteFilter, [setFilters]);

  function handleChangeOperator() {
    setOperator(prevOperator => (prevOperator === "AND" ? "OR" : "AND"));
  }

  const memorizedHandleChangeOperator = React.useCallback(handleChangeOperator, [setOperator]);

  return (
    <Filter
      onChange={memorizedHandleChange}
      onAddFilter={memorizedAddFilter}
      onDeleteFilter={memorizedDeleteFilter}
      filters={filters}
      columns={columns}
      operator={operator}
      onChangeOperator={memorizedHandleChangeOperator}
    />
  );
}
