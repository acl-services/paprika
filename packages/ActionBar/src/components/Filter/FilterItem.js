import React from "react";
import PropTypes from "prop-types";
import Button from "@paprika/button";
import useI18n from "@paprika/l10n/lib/useI18n";
import DatePicker from "./DatePicker";
import InlineSelect from "../InlineSelect/InlineSelect";
import InlineInput from "../InlineInput/InlineInput";
import rules, { localeKeysByRule } from "./rules";
import FilterContext from "./context";
import { columnTypes, logicalFilterOperators, changeTypes } from "../../constants";
import * as sc from "./Filter.styles";

const propTypes = {
  columnId: PropTypes.string.isRequired,
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  index: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
  renderValueField: PropTypes.func,
  rule: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]).isRequired,
};

const defaultProps = {
  id: null,
  renderValueField: null,
};

function FilterItem(props) {
  const {
    index,
    onChange,
    onDelete,
    id,
    columnId: selectedColumnId,
    rule: selectedRule,
    value,
    renderValueField: renderCustomValueField,
  } = props;
  const { columns, filtersRef, onChangeOperator, operator, rulesByType } = React.useContext(FilterContext);
  const I18n = useI18n();

  const selectedColumnType = columns.find(({ id }) => id === selectedColumnId).type;

  function handleRemoveFilter() {
    filtersRef.current.focus();
    onDelete(id);
  }

  function handleChangeColumn(event) {
    const newColumnId = event.target.value;
    onChange(changeTypes.COLUMN, { id, columnId: newColumnId });
  }

  function handleChangeRule(event) {
    onChange(changeTypes.RULE, { id, rule: event.target.value });
  }

  function handleChangeValue(newValue) {
    onChange(changeTypes.FILTER_VALUE, { id, value: newValue });
  }

  function handleChangeDatePicker(momentDate) {
    filtersRef.current.focus();

    if (momentDate === null) {
      handleChangeValue("");
    } else {
      handleChangeValue(momentDate.format(columns.find(({ id }) => id === selectedColumnId).momentParsingFormat));
    }
  }

  function handleChangeBooleanFilterValue(event) {
    onChange(changeTypes.FILTER_VALUE, { id, value: event.target.value === "true" });
  }

  function renderRuleField() {
    switch (selectedColumnType) {
      case columnTypes.BOOLEAN:
        return I18n.t("actionBar.filter.rules.is");
      default:
        return (
          <InlineSelect
            onChange={handleChangeRule}
            value={selectedRule}
            selectedLabel={I18n.t(`actionBar.filter.rules.${localeKeysByRule[selectedRule]}`)}
            data-pka-anchor="actionBar.filter.ruleSelector"
          >
            {rulesByType[selectedColumnType].map(rule => (
              <option key={rule} value={rule}>
                {I18n.t(`actionBar.filter.rules.${localeKeysByRule[rule]}`)}
              </option>
            ))}
          </InlineSelect>
        );
    }
  }

  function renderValueField() {
    const shouldNotShowValueField =
      selectedRule === rules.IS_BLANK ||
      selectedRule === rules.IS_NOT_BLANK ||
      selectedRule === rules.IS_EMPTY ||
      selectedRule === rules.IS_NOT_EMPTY;

    if (shouldNotShowValueField) return null;

    if (renderCustomValueField) return renderCustomValueField();

    switch (selectedColumnType) {
      case columnTypes.BOOLEAN:
        return (
          <InlineSelect
            value={`${value}`}
            onChange={handleChangeBooleanFilterValue}
            selectedLabel={I18n.t(`actionBar.filter.rules.${value}`)}
          >
            <option value="true">{I18n.t("actionBar.filter.rules.true")}</option>
            <option value="false">{I18n.t("actionBar.filter.rules.false")}</option>
          </InlineSelect>
        );
      case columnTypes.DATE:
        return (
          <DatePicker
            initialDate={value}
            onChange={handleChangeDatePicker}
            parsingFormat={columns.find(({ id }) => id === selectedColumnId).momentParsingFormat}
          />
        );
      default:
        return (
          <InlineInput inputKey={`${selectedColumnId}-${selectedRule}`} value={value} onChange={handleChangeValue} />
        );
    }
  }

  function renderPrefix() {
    const staticPrefix = I18n.t(`actionBar.filter.${operator === logicalFilterOperators.AND ? "and" : "or"}`);

    switch (index) {
      case 0:
        return I18n.t("actionBar.filter.where");
      case 1:
        if (onChangeOperator) {
          return (
            <InlineSelect
              onChange={onChangeOperator}
              value={operator}
              selectedLabel={
                operator === logicalFilterOperators.AND ? I18n.t("actionBar.filter.and") : I18n.t("actionBar.filter.or")
              }
            >
              <option value={logicalFilterOperators.AND}>{I18n.t("actionBar.filter.and")}</option>
              <option value={logicalFilterOperators.OR}>{I18n.t("actionBar.filter.or")}</option>
            </InlineSelect>
          );
        }
        return staticPrefix;
      default:
        return staticPrefix;
    }
  }

  return (
    <sc.FilterItem data-pka-anchor="actionBar.filter.filterItem">
      <Button.Close data-pka-anchor="actionBar.filter.deleteFilterButton" onClick={handleRemoveFilter} size="small" />
      {renderPrefix()}
      <InlineSelect
        onChange={handleChangeColumn}
        value={selectedColumnId}
        selectedLabel={columns.find(column => column.id === selectedColumnId).label}
      >
        {columns.map(column => (
          <option key={column.id} value={column.id}>
            {column.label}
          </option>
        ))}
      </InlineSelect>

      {renderRuleField()}
      {renderValueField()}
    </sc.FilterItem>
  );
}

FilterItem.propTypes = propTypes;
FilterItem.defaultProps = defaultProps;

export default React.memo(FilterItem);
