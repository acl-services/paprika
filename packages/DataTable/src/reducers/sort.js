import sort from "../helpers/sort";
import { actions } from "../constants";

export default function sortReducer(state, action) {
  if (action.type === actions.SORT)
    return {
      ...action.changes,
      sortColumn: action.payload.columnId,
      sortDirection: action.payload.direction,
      sortedOrder: sort({
        data: action.changes.data,
        columnId: action.payload.columnId,
        direction: action.payload.direction,
      }).map(item => item[state.keygen]),
    };

  return action.changes;
}
