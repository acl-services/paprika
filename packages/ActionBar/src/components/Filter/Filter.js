import React from "react";
import PropTypes from "prop-types";
import Popover from "@paprika/popover";
import Button from "@paprika/button";
import useI18n from "@paprika/l10n/lib/useI18n";
import CheckIcon from "@paprika/icon/lib/Check";
import FilterItem from "./FilterItem";
import { rules, defaultRulesByType } from "./rules";
import FilterContext from "./context";
import { logicalFilterOperators } from "../../constants";
import columnShape from "../../columnShape";

import * as sc from "./Filter.styles";
import { GenericNoAppliedPlaceholder } from "../../ActionBar.styles";

const propTypes = {
  appliedNumber: PropTypes.number,
  children: PropTypes.node,
  columns: PropTypes.arrayOf(PropTypes.shape(columnShape)).isRequired,
  onAddFilter: PropTypes.func.isRequired,
  onApply: PropTypes.func.isRequired,
  onCancel: PropTypes.func,
  onChangeOperator: PropTypes.func,
  onClose: PropTypes.func,
  onOpen: PropTypes.func,
  operator: PropTypes.oneOf([logicalFilterOperators.AND, logicalFilterOperators.OR]),
  rulesByType: PropTypes.objectOf(PropTypes.arrayOf(PropTypes.objectOf(rules))),
};

const defaultProps = {
  appliedNumber: 0,
  children: null,
  onChangeOperator: null,
  onCancel: () => {},
  onClose: () => {},
  onOpen: () => {},
  operator: logicalFilterOperators.AND,
  rulesByType: defaultRulesByType,
};

function getLabelText(numberOfFilters, I18n) {
  switch (numberOfFilters) {
    case 0:
      return I18n.t("actionBar.filter.label");
    case 1:
      return I18n.t("actionBar.filter.singular_label");
    default:
      return I18n.t("actionBar.filter.plural_label", { numberOfFilters });
  }
}

export default function Filter(props) {
  const {
    appliedNumber,
    children,
    columns,
    onAddFilter,
    onApply,
    onCancel,
    onChangeOperator,
    onClose,
    onOpen,
    operator,
    rulesByType,
  } = props;
  const I18n = useI18n();
  const filtersRef = React.useRef(null);
  const [isOpen, setIsOpen] = React.useState(false);

  function handleClickTrigger() {
    setIsOpen(prevIsOpen => {
      if (prevIsOpen) return false;

      onOpen();
      return true;
    });
  }

  function handleClose() {
    setIsOpen(false);
    onClose();
  }

  function handleApply() {
    setIsOpen(false);
    onApply();
  }

  function handleCancel() {
    setIsOpen(false);
    onCancel();
  }

  return (
    <FilterContext.Provider value={{ filtersRef, columns, operator, onChangeOperator, rulesByType }}>
      <Popover align="bottom" edge="left" maxWidth={600} offset={8} isOpen={isOpen} onClose={handleClose}>
        <sc.Trigger
          isSemantic={false}
          kind="flat"
          onClick={handleClickTrigger}
          hasFilterApplied={appliedNumber > 0}
          isOpen={isOpen}
        >
          <sc.Icon />
          {getLabelText(appliedNumber, I18n)}
        </sc.Trigger>
        <Popover.Content>
          <Popover.Card>
            <sc.FiltersPanel ref={filtersRef} tabIndex={-1}>
              {React.Children.count(children) === 0 ? (
                <GenericNoAppliedPlaceholder>
                  {I18n.t("actionBar.filter.no_filters_applied")}
                </GenericNoAppliedPlaceholder>
              ) : (
                children
              )}
            </sc.FiltersPanel>
            <sc.Footer>
              <Button onClick={onAddFilter} kind="minor" data-pka-anchor="actionBar.filter.addFilterButton">
                {I18n.t(`actionBar.filter.add_filter`)}
              </Button>
              <Button onClick={handleApply} kind="flat" icon={<CheckIcon />}>
                {I18n.t("actions.apply")}
              </Button>
              <Button onClick={handleCancel} kind="minor">
                {I18n.t("actions.cancel")}
              </Button>
            </sc.Footer>
          </Popover.Card>
        </Popover.Content>
        <Popover.Tip />
      </Popover>
    </FilterContext.Provider>
  );
}

Filter.propTypes = propTypes;
Filter.defaultProps = defaultProps;
Filter.displayName = "ActionBar.Filter";
Filter.defaultRulesByType = defaultRulesByType;
Filter.rules = rules;
Filter.Item = FilterItem;
