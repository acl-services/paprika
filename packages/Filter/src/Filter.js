import React from "react";
import PropTypes from "prop-types";
import AddIcon from "@paprika/icon/lib/Add";
import Button from "@paprika/button";
import FilterIcon from "@paprika/icon/lib/Filter";
import Panel from "@paprika/panel";
import useI18n from "@paprika/l10n/lib/useI18n";
import FilterItem from "./components/FilterItem";
import FilterContext from "./context";
import columnShape from "./columnShape";
import * as types from "./types";
import rules, { defaultRulesByType, logicalFilterOperators } from "./rules";
import isEqualityRule from "./helpers/isEqualityRule";
import * as sc from "./Filter.styles";

export default function Filter({
  children,
  columns,
  data,
  numberApplied,
  maxFiltersAllowed,
  onAddFilter,
  onApply,
  onCancel,
  onChangeOperator,
  onClear,
  operator,
  rulesByType,
  zIndex,
  ...moreProps
}) {
  const I18n = useI18n();
  const filterRef = React.useRef(null);
  const [isOpen, setIsOpen] = React.useState(false);

  function handleClickTrigger() {
    setIsOpen(prevIsOpen => !prevIsOpen);
  }

  function handleApply() {
    onApply();
    setIsOpen(false);
  }

  function handleCancel() {
    setIsOpen(false);
    onCancel();
  }

  function handleClear() {
    onClear();
  }

  const childrenArray = React.Children.toArray(children);

  const allFieldsAreAlreadyFilteredBy =
    operator === logicalFilterOperators.AND &&
    childrenArray.length === columns.length &&
    childrenArray.every(child => isEqualityRule(child.props.rule));

  return (
    <FilterContext.Provider value={{ filterRef, children, columns, data, operator, onChangeOperator, rulesByType }}>
      <Panel
        a11yText={I18n.t("filter.label")}
        data-pka-anchor="filter.panel"
        isCompact
        isOpen={isOpen}
        onClose={handleCancel}
        zIndex={zIndex}
        {...moreProps}
      >
        <Panel.Header>
          <sc.PanelHeaderWrapper>
            <FilterIcon aria-hidden="true" />
            {I18n.t("filter.label")}
          </sc.PanelHeaderWrapper>
        </Panel.Header>

        <Panel.Trigger
          icon={<FilterIcon />}
          isSemantic={false}
          kind={Button.types.kind.FLAT}
          onClick={handleClickTrigger}
        >
          {numberApplied > 0 ? I18n.t("filter.label_with_count", { count: numberApplied }) : I18n.t("filter.label")}
        </Panel.Trigger>
        <sc.FilterPanel ref={filterRef}>
          {childrenArray.length === 0 ? (
            <sc.GenericNoAppliedPlaceholder>{I18n.t("filter.no_filters_applied")}</sc.GenericNoAppliedPlaceholder>
          ) : (
            children
          )}
          {childrenArray.length < maxFiltersAllowed && !allFieldsAreAlreadyFilteredBy && (
            <Button data-pka-anchor="filter.addFilterButton" icon={<AddIcon />} onClick={onAddFilter}>
              {I18n.t("filter.actions.add")}
            </Button>
          )}
        </sc.FilterPanel>
        <Panel.Footer isSticky>
          <sc.Footer>
            <Button data-pka-anchor="filter.applyButton" onClick={handleApply} kind="primary">
              {I18n.t("actions.apply")}
            </Button>
            <Button data-pka-anchor="filter.cancelButton" onClick={handleCancel} kind="minor">
              {I18n.t("actions.cancel")}
            </Button>
            <Button data-pka-anchor="filter.clearButton" onClick={handleClear} kind="minor">
              {I18n.t("filter.actions.clear")}
            </Button>
          </sc.Footer>
        </Panel.Footer>
      </Panel>
    </FilterContext.Provider>
  );
}

Filter.displayName = "Filter";

Filter.defaultRulesByType = defaultRulesByType;
Filter.Item = FilterItem;
Filter.operator = logicalFilterOperators;
Filter.rules = rules;
Filter.types = types;

const propTypes = {
  numberApplied: PropTypes.number,
  children: PropTypes.node,
  columns: PropTypes.arrayOf(PropTypes.shape(columnShape)).isRequired,
  data: PropTypes.arrayOf(PropTypes.shape({})),
  maxFiltersAllowed: PropTypes.number,
  onAddFilter: PropTypes.func.isRequired,
  onApply: PropTypes.func.isRequired,
  onCancel: PropTypes.func,
  onChangeOperator: PropTypes.func,
  onClear: PropTypes.func,
  operator: PropTypes.oneOf([Filter.operator.AND, Filter.operator.OR]),
  rulesByType: PropTypes.objectOf(PropTypes.arrayOf(PropTypes.oneOf(Object.values(Filter.rules)))),
  zIndex: PropTypes.number,
};

const defaultProps = {
  children: null,
  data: null,
  maxFiltersAllowed: 9999,
  numberApplied: 0,
  onCancel: () => {},
  onChangeOperator: null,
  onClear: () => {},
  operator: Filter.operator.AND,
  rulesByType: Filter.defaultRulesByType,
  zIndex: undefined,
};

Filter.propTypes = propTypes;
Filter.defaultProps = defaultProps;
