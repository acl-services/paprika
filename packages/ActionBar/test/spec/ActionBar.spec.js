import testers from "../../src/hooks/useFilter/testers";
import Filter from "../../src/components/Filter/Filter";

import sort from "../../src/hooks/useSort/sort";
import sampleData from "../data";

describe("ActionBar Filter", () => {
  it("should run filter rules IS NOT", () => {
    expect(testers[Filter.rules.IS_NOT]("value", "")).toBe(true);
    expect(testers[Filter.rules.IS_NOT]("value", "testValue")).toBe(true);
    expect(testers[Filter.rules.IS_NOT]("value", "value")).toBe(false);
  });

  it("should run filter rules DOES NOT CONTAIN", () => {
    expect(testers[Filter.rules.DOES_NOT_CONTAIN]("value", "test value")).toBe(true);
    expect(testers[Filter.rules.DOES_NOT_CONTAIN]("test VaLue", "value")).toBe(false);
  });

  it("should run filter rules IS NOT BLANK", () => {
    expect(testers[Filter.rules.IS_NOT_BLANK]("test value")).toBe(true);
    expect(testers[Filter.rules.IS_NOT_BLANK]("")).toBe(false);
  });
  it("should run filter rules IS NOT EQUAL TO", () => {
    expect(testers[Filter.rules.NOT_EQUAL_TO](100, 200)).toBe(true);
    expect(testers[Filter.rules.NOT_EQUAL_TO](100, 100)).toBe(false);
  });

  it("should run filter rules IS GREATER THAN OR EQUAL TO ", () => {
    expect(testers[Filter.rules.GREATER_THAN_OR_EQUAL_TO](200, 200)).toBe(true);
    expect(testers[Filter.rules.GREATER_THAN_OR_EQUAL_TO](250, 200)).toBe(true);
    expect(testers[Filter.rules.GREATER_THAN_OR_EQUAL_TO](150, 200)).toBe(false);
  });
  it("should run filter rules IS LESS THAN OR EQUAL TO ", () => {
    expect(testers[Filter.rules.LESS_THAN_OR_EQUAL_TO](200, 200)).toBe(true);
    expect(testers[Filter.rules.LESS_THAN_OR_EQUAL_TO](150, 200)).toBe(true);
    expect(testers[Filter.rules.LESS_THAN_OR_EQUAL_TO](250, 200)).toBe(false);
  });
  it("should run filter rules IS NOT EMPTY", () => {
    expect(testers[Filter.rules.IS_NOT_BLANK]("test value")).toBe(true);
    expect(testers[Filter.rules.IS_NOT_BLANK]("")).toBe(false);
  });
  it("should run filter rules IS AFTER", () => {
    expect(testers[Filter.rules.IS_AFTER]("10/08/2020", "10/02/2020", { momentParsingFormat: "MM/DD/YYYY" })).toBe(
      true
    );
    expect(testers[Filter.rules.IS_AFTER]("10/02/2020", "10/08/2020", { momentParsingFormat: "MM/DD/YYYY" })).toBe(
      false
    );
  });
});

describe("ActionBar Sort", () => {
  it("should sort by name STRING ASC", () => {
    expect(sort({ data: sampleData, columnId: "name", direction: "ASCEND", columnType: "TEXT" })).toMatchSnapshot();
  });

  it("should sort by name STRING DESC", () => {
    expect(
      sort({ data: sampleData, columnId: "shareable", direction: "DESCEND", columnType: "TEXT" })
    ).toMatchSnapshot();
  });

  it("should sort by goals NUMBER ASC", () => {
    expect(sort({ data: sampleData, columnId: "goals", direction: "ASCEND", columnType: "NUMBER" })).toMatchSnapshot();
  });

  it("should sort by goals NUMBER DESC", () => {
    expect(sort({ data: sampleData, columnId: "goals", direction: "DESCEND", columnType: "NUMBER" })).toMatchSnapshot();
  });

  it("should sort by joined by DATE ASC", () => {
    expect(
      sort({
        data: sampleData,
        columnId: "joined",
        direction: "ASCEND",
        columnType: "DATE",
        momentParsingFormat: "MM/DD/YYYY",
      })
    ).toMatchSnapshot();
  });

  it("should sort by joined by DATE DESC", () => {
    expect(
      sort({
        data: sampleData,
        columnId: "joined",
        direction: "DESCEND",
        columnType: "DATE",
        momentParsingFormat: "MM/DD/YYYY",
      })
    ).toMatchSnapshot();
  });
});
