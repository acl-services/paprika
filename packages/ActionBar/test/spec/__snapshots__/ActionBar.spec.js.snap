// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar Sort should sort by goals NUMBER ASC 1`] = `
Array [
  Object {
    "country": "Brazil",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 767,
    "id": "3",
    "joined": "04/01/2014",
    "level": "mid",
    "link": "https://wegalvanize.com",
    "name": "Pelé",
    "shareable": true,
    "status": "inactive",
  },
  Object {
    "country": "Brazil",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 772,
    "id": "2",
    "joined": "08/11/2019",
    "level": "high",
    "link": "https://wegalvanize.com",
    "name": "Romário",
    "shareable": false,
    "status": "inactive",
  },
  Object {
    "country": "Austria",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 805,
    "id": "1",
    "joined": "12/12/2019",
    "level": "low",
    "link": "https://wegalvanize.com",
    "name": "Josef Bican ‡",
    "shareable": true,
    "status": "inactive",
  },
]
`;

exports[`ActionBar Sort should sort by goals NUMBER DESC 1`] = `
Array [
  Object {
    "country": "Austria",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 805,
    "id": "1",
    "joined": "12/12/2019",
    "level": "low",
    "link": "https://wegalvanize.com",
    "name": "Josef Bican ‡",
    "shareable": true,
    "status": "inactive",
  },
  Object {
    "country": "Brazil",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 772,
    "id": "2",
    "joined": "08/11/2019",
    "level": "high",
    "link": "https://wegalvanize.com",
    "name": "Romário",
    "shareable": false,
    "status": "inactive",
  },
  Object {
    "country": "Brazil",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 767,
    "id": "3",
    "joined": "04/01/2014",
    "level": "mid",
    "link": "https://wegalvanize.com",
    "name": "Pelé",
    "shareable": true,
    "status": "inactive",
  },
]
`;

exports[`ActionBar Sort should sort by joined by DATE ASC 1`] = `
Array [
  Object {
    "country": "Brazil",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 767,
    "id": "3",
    "joined": "04/01/2014",
    "level": "mid",
    "link": "https://wegalvanize.com",
    "name": "Pelé",
    "shareable": true,
    "status": "inactive",
  },
  Object {
    "country": "Brazil",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 772,
    "id": "2",
    "joined": "08/11/2019",
    "level": "high",
    "link": "https://wegalvanize.com",
    "name": "Romário",
    "shareable": false,
    "status": "inactive",
  },
  Object {
    "country": "Austria",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 805,
    "id": "1",
    "joined": "12/12/2019",
    "level": "low",
    "link": "https://wegalvanize.com",
    "name": "Josef Bican ‡",
    "shareable": true,
    "status": "inactive",
  },
]
`;

exports[`ActionBar Sort should sort by joined by DATE DESC 1`] = `
Array [
  Object {
    "country": "Austria",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 805,
    "id": "1",
    "joined": "12/12/2019",
    "level": "low",
    "link": "https://wegalvanize.com",
    "name": "Josef Bican ‡",
    "shareable": true,
    "status": "inactive",
  },
  Object {
    "country": "Brazil",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 772,
    "id": "2",
    "joined": "08/11/2019",
    "level": "high",
    "link": "https://wegalvanize.com",
    "name": "Romário",
    "shareable": false,
    "status": "inactive",
  },
  Object {
    "country": "Brazil",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 767,
    "id": "3",
    "joined": "04/01/2014",
    "level": "mid",
    "link": "https://wegalvanize.com",
    "name": "Pelé",
    "shareable": true,
    "status": "inactive",
  },
]
`;

exports[`ActionBar Sort should sort by name STRING ASC 1`] = `
Array [
  Object {
    "country": "Austria",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 805,
    "id": "1",
    "joined": "12/12/2019",
    "level": "low",
    "link": "https://wegalvanize.com",
    "name": "Josef Bican ‡",
    "shareable": true,
    "status": "inactive",
  },
  Object {
    "country": "Brazil",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 767,
    "id": "3",
    "joined": "04/01/2014",
    "level": "mid",
    "link": "https://wegalvanize.com",
    "name": "Pelé",
    "shareable": true,
    "status": "inactive",
  },
  Object {
    "country": "Brazil",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 772,
    "id": "2",
    "joined": "08/11/2019",
    "level": "high",
    "link": "https://wegalvanize.com",
    "name": "Romário",
    "shareable": false,
    "status": "inactive",
  },
]
`;

exports[`ActionBar Sort should sort by name STRING DESC 1`] = `
Array [
  Object {
    "country": "Austria",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 805,
    "id": "1",
    "joined": "12/12/2019",
    "level": "low",
    "link": "https://wegalvanize.com",
    "name": "Josef Bican ‡",
    "shareable": true,
    "status": "inactive",
  },
  Object {
    "country": "Brazil",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 767,
    "id": "3",
    "joined": "04/01/2014",
    "level": "mid",
    "link": "https://wegalvanize.com",
    "name": "Pelé",
    "shareable": true,
    "status": "inactive",
  },
  Object {
    "country": "Brazil",
    "description": "Puppy kitty ipsum dolor sit good dog foot stick canary. Teeth Mittens furry treats fish.",
    "goals": 772,
    "id": "2",
    "joined": "08/11/2019",
    "level": "high",
    "link": "https://wegalvanize.com",
    "name": "Romário",
    "shareable": false,
    "status": "inactive",
  },
]
`;
